### DISCUSSION TITLE
Changing container within a Function App

### DISCUSSION THREAD

**Discussion by jorg-snyders** [IEvangelist/azure-cosmos-dotnet-repository#104] [Q&A] [answered]
Hi,

I was wondering if it is possible to change the containerId when in a different function. 
Right now I have it setup with the RepositoryOptions in the App settings, which works great. 
But one of my functions requires me to fetch data from another container, so I would like to be able to switch the containerId in that function. 

Is this functionality possible or should I create an entire new function app for that one function?

Thanks!

**Comment by mumby0168** [comment, marked as answer]
> Hi,
> 
> I was wondering if it is possible to change the containerId when in a different function.
> Right now I have it setup with the RepositoryOptions in the App settings, which works great.
> But one of my functions requires me to fetch data from another container, so I would like to be able to switch the containerId in that function.
> 
> Is this functionality possible or should I create an entire new function app for that one function?
> 
> Thanks!

Hey there,

Yes so we do support this on the `RepositoryOptions` there is a property called `ContainerPerItemType` if you set this to true you can then use attributes or config to tell the library which container and partition key to use for each instance of an `IItem`. If you checkout the samples I think the `ServiceTier` & the `AzureFunctionsTier` shows examples of this. See [here](https://github.com/IEvangelist/azure-cosmos-dotnet-repository/blob/main/Microsoft.Azure.CosmosRepository.Samples/AzureFunctionTier/Startup.cs)

**Reply by jorg-snyders** [reply]
Hey mumby,

Thanks for this! This was exactly what I was looking for!
